Cert branch

gendjo-owlhead opened the pull request:
Please review

gmustafov-tricentis committed:
Add complete certification project to cert_branch

gmustafov-tricentis committed:
Add GitHub configuration files and workflows

Commit:
chore: fix Playwright runner issues\n\n- Switch to ESM (type: module, nodenext)\n- Fix Faker ESM imports\n- Fix JSON import with assert\n- Ensure single @playwright/test instance

Commit:
test(visual): add macOS baselines for profile visual tests

gendjo-owlhead committed:
Update README.md

Commit:
final changes

Commit:
Polished version

fifkapetr commented:
# Hodnocení certifikačního projektu

**Student:** Gendžaj Mustafov
**GitHub:** [gendjo-owlhead/Certifikace/pull/3](https://github.com/gendjo-owlhead/Certifikace/pull/3?authuser=0)
**Certifikace:** Playwright Certification
**Datum hodnocení:** 30. 9. 2025 8:04:06
**Hodnotitel:** Petr Fifka

## Celkové hodnocení

- **Celkový počet bodů:** 1125/1500
- **Procentuální úspěšnost:** 75%
- **Výsledek:** ✅ PROŠEL

- **Kritérium úspěchu:** 75% (1125 bodů)

## Podrobné hodnocení

### Splnění zadání
**Celkem:** 420/500 bodů

- **E2E test:** 125/125 bodů
- **API:** 75/75 bodů
- **DDT:** 70/125 bodů
  - *Poznámka:* - Špatně vyřešený bug (500) - uděláno, aby test prošel, ale ve skutečnosti, má spadnout, protože jinak vytváříme tzv. syndrom věčně zeleného testu (nízká účinnost nálezu chyb). Správná cesta by byla tyto kombinace přeskočit.
- Zbytečně komplexní implementace selektorů a kontrol - dá se udělat jednoduše pomocí několika řádků a není třeba vymýšlet over-engineering řešení
- **Atomické testy:** 75/100 bodů
  - *Poznámka:* - Chybí testy na labely profilu/accountu
- **Vizuální test:** 75/75 bodů

### Spustitelnost
**Celkem:** 125/150 bodů

- **Aplikace se spustí bez chyb:** 125/150 bodů
  - *Poznámka:* - TS syntaktické chyby: špatné importy (bez .ts) a špatně nastavený tsconfig (chybějící: allowImportingTsExtensions, noEmit) a package.json (type: module)

### Technická zdatnost
**Celkem:** 285/500 bodů

- **Page Objecty:** 30/80 bodů
  - *Poznámka:* - Některé prvky nejsou v PO (E2E testy) - expecty, kroky, lokátory
- API objekty nejsou implementovány
- Test stepy by mohly být v objektech
- **Fluent API:** 25/70 bodů
  - *Poznámka:* - Fluent api je částečně implementován, ale blokován předchozími nedostatky
- **Udržitelná citlivá data:** 0/60 bodů
  - *Poznámka:* - Citlivá data nahardcodována v testech
- **DDT v JSON:** 70/70 bodů
- **Asserty:** 35/60 bodů
  - *Poznámka:* - Kontrola probíhá často na úrovni lokátoru než konkrétního assertu, což může způsobovat celou řadu problémů (stabilita, analysis hell v debuggingu, problémy ve vícejazyčných app..)
- **Generátory, externí knihovny:** 70/70 bodů
- **Selektory:** 25/40 bodů
  - *Poznámka:* - Lokátory se často chytají na texty namísto na data-testid i když jsou ve většině případů dostupná.
- **Struktura testů (describe, test, before...):** 30/50 bodů
  - *Poznámka:* - Díky chybějícím objektům je špatná orientace v testech
- Kombinace funkcionálního a objektového programování (pomocné funkce například pro saveProfile) ztěžují orientaci ve frameworku - tyto funkce by měly být ideálně implementovány v PO

### Čistota kódu
**Celkem:** 150/200 bodů

- **Není hardcode:** 20/35 bodů
  - *Poznámka:* - Implicitní timeout (3000 ms), opakující se - zpomalování testů
- Podmínky v testech - nejsou jednoznačné - testy i lokátory obsahují větvení
- **Umístění souborů:** 35/35 bodů
- **Pojmenovávání dává smysl:** 30/30 bodů
- **Konvence názvů dodržena:** 30/30 bodů
- **Nejsou duplicity:** 20/35 bodů
  - *Poznámka:* - Duplicity se objevují ve všech testech (login, url...)
- **Kód je čitelný:** 15/35 bodů
  - *Poznámka:* - Díky over-engineering řešením je čitelnost testů malá (ddt fory a logika, E2E funkce)

### Git
**Celkem:** 150/150 bodů

- **Struktura (branch, gitignore, PR):** 75/75 bodů
- **Názvosloví:** 75/75 bodů

## Bonus/Malus položky

- **Nalezení chyb:** 0 bodů
- **Bonus/Malus:** -5 bodů

## Poznámky

**BONUS**
- Datové objekty +10b
- Axe implementace +15b

**MALUS**
- Velké množství implicitních čekání -30

---
*Exportováno: 30. 9. 2025 8:05:55*

Commit:
Refactor tests to use TEST_IDS for improved maintainability and readability

- Updated dashboard-content.spec.ts to utilize TEST_IDS for locators and assertions.
- Refactored dashboard-header.spec.ts to replace hardcoded selectors with TEST_IDS.
- Modified dashboard-menu.spec.ts to implement TEST_IDS for navigation items and assertions.
- Enhanced account-balances.spec.ts to use TEST_IDS for account balance verification.
- Revised complete-user-flow.spec.ts to incorporate TEST_IDS for profile and account interactions.
- Improved profile-visual.spec.ts to utilize TEST_IDS for visual testing of profile elements.
- Introduced test-ids.ts to centralize test ID management for easier updates and consistency across tests.
